base/path.jl

```diff
@@ -248,16 +248,15 @@ function splitpath(p::String)
     return out
 end
 
-joinpath(path::AbstractString)::String = path
-
 if Sys.iswindows()
 
-function joinpath(path::AbstractString, paths::AbstractString...)::String
-    result_drive, result_path = splitdrive(path)
+function joinpath(paths)::String
+    isempty(paths) && return path_separator
+    result_drive, result_path = splitdrive(paths[1])
 
-    local p_drive, p_path
-    for p in paths
-        p_drive, p_path = splitdrive(p)
+    p_path = ""
+    for i in 2:length(paths)
```

Comment:
```suggestion
    for i in firstindex(paths)+1:lastindex(paths)
```
Since we're purporting to support any kind of abstract vector, technically we need this.